### ISSUE TITLE
salt.states.grains.list_present should not show changes if none are made!

### ISSUE THREAD

**Issue by sjorge** [saltstack/salt#30651]
```
          ID: package:install:salt::completed
    Function: grains.list_present
        Name: packages:salt
      Result: True
     Comment: Append value 2016.1.2-dev to grain packages:salt
     Started: 23:21:55.939789
    Duration: 4.641 ms
     Changes:
              ----------
              new:
                  The val 2016.1.2-dev was already in the list packages:salt
```

Nothing was actually! ```changes``` should be empty, the ```comment``` should probably be what is in changes.new at the moment.

2016.1.2-dev was present in the list, so the state = OK, no changes made.

**Comment by rallytime** [comment]
@sjorge Good find. Your "`changes` should be empty, the `comment` should probably be what is in changes.new at the moment" statement is correct.

**Comment by abednarik** [comment]
I tried to reproduce this one in develop and 2015.8 without luck. I always get this output

```bash
ubuntu:
----------
          ID: roles
    Function: grains.list_present
      Result: True
     Comment: Value web is already in grain roles
     Started: 00:36:46.774447
    Duration: 0.496 ms
     Changes:

Summary for ubuntu
------------
Succeeded: 1
Failed:    0
------------
Total states run:     1
```

**Comment by rallytime** [comment]
Thanks for trying to reproduce, @abednarik.

@sjorge Can you post your `salt --versions-report` here, please?

**Comment by sjorge** [comment]
I'm using develop, build from yesterday

**Comment by abednarik** [comment]
Hi @sjorge 

 Can you share what you are doing? I cannot reproduce the issue.

Thanks.

**Comment by sjorge** [comment]
I have 2 states install and uninstall that install a custom software package.

At the end of install I set a grain:
```
package:install:{{ pkg.name }}::completed:  ## update grain on completion
  grains.list_present:
    - name: packages:{{ pkg.name }}
    - value: {{ pkg.version }}
    - order: last
```
The packages are versioned so multiple ones can be installed.

In the uninstall state I remove the grain:
```
package:uninstall:{{ pkg.name }}::completed:  ## remove grain
  grains.absent:
    - name: packages:{{ pkg.name }}
    - destructive: true
    - order: last
```

Resulting in a grain looking something like this
```
packages:
  pkg1:
    - 1.0
    - 1.1
  pkg2:
    - 1.2
    - 2.0
```

Hover if install is ran a second time for a version (it skips all steps as expected because they are in the correct state). grains.list_present will always report a change, even if the value is already in the list.

**Comment by sjorge** [comment]
# reproduction steps (develop from yesterday)

Write a simple state, execute it and check the grain

```yaml
package_set_grain:
  grains.list_present:
    - name: packages:test
    - value: 1.0
```

```bash
[root@cronos /srv/salt/states/package/salt]# salt-call state.sls test
local:
----------
          ID: package_set_grain
    Function: grains.list_present
        Name: packages:test
      Result: True
     Comment: Append value 1.0 to grain packages:test
     Started: 23:50:30.099874
    Duration: 517.006 ms
     Changes:
              ----------
              new:
                  ----------
                  packages:
                      ----------
                      test:
                          - 1.0

Summary for local
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
```

```bash
[root@cronos /srv/salt/states/package/salt]# salt-call grains.get packages
local:
    ----------
    test:
        - 1.0
```

**Up till now it all looks good**

Execute the state again.

```bash
[root@cronos /srv/salt/states/package/salt]# salt-call state.sls test
local:
----------
          ID: package_set_grain
    Function: grains.list_present
        Name: packages:test
      Result: True
     Comment: Append value 1.0 to grain packages:test
     Started: 23:54:14.505104
    Duration: 3.342 ms
     Changes:
              ----------
              new:
                  The val 1.0 was already in the list packages:test

Summary for local
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
```
**That's odd! There is one new change! But nothing happened so it should not be marked so**

**Comment by sjorge** [comment]
It only happens when using a nested grain!

**Comment by rallytime** [comment]
Nice tracking that one down!

**Comment by sjorge** [comment]
Just a heads up as this did not get merged into 2016.3, the issue is still there!
